packages/components/src/Image/tests/jest/Image.test.tsx

```diff
@@ -0,0 +1,56 @@
+import { render, screen } from "@hopper-ui/test-utils";
+import { createRef } from "react";
+
+import { Image } from "../../src/Image.tsx";
+import { ImageContext } from "../../src/ImageContext.ts";
+
+describe("Image", () => {
+    it("should render with default class", () => {
+        render(<Image data-testid="Image" src="https://i.pravatar.cc/96?img=1" />);
+
+        const element = screen.getByTestId("Image");
+        expect(element).toHaveClass("hop-Image");
+    });
+
+    it("should support custom class", () => {
+        render(<Image data-testid="Image" className="test" src="https://i.pravatar.cc/96?img=1" />);
+
+        const element = screen.getByTestId("Image");
+        expect(element).toHaveClass("hop-Image");
+        expect(element).toHaveClass("test");
+    });
+
+    it("should support custom style", () => {
+        render(<Image data-testid="Image" marginTop="stack-sm" style={{ marginBottom: "13px" }} src="https://i.pravatar.cc/96?img=1" />);
+
+        const element = screen.getByTestId("Image");
+        expect(element).toHaveStyle({ marginTop: "var(--hop-space-stack-sm)", marginBottom: "13px" });
+    });
+
+    it("should support DOM props", () => {
+        render(<Image data-testid="Image" data-foo="bar" src="https://i.pravatar.cc/96?img=1" />);
+
+        const element = screen.getByTestId("Image");
+        expect(element).toHaveAttribute("data-foo", "bar");
+    });
+
+    it("should support slots", () => {
+        render(
+            <ImageContext.Provider value={{ slots: { test: { "aria-label": "test" } } }}>
+                <Image data-testid="Image" slot="test" src="https://i.pravatar.cc/96?img=1" />
+            </ImageContext.Provider>
+        );
+
+        const element = screen.getByTestId("Image");
```

Comment:
i know there serie of tests is copy pasted so no worry, just an FYI, i always refer to this list of priority of what to query: https://testing-library.com/docs/queries/about/#priority

getByTestId should always be the "last resort"